components/tidb_query_datatype/src/def/field_type.rs

```diff
@@ -145,20 +146,25 @@ impl fmt::Display for Collation {
     }
 }
 
-#[derive(Debug, Clone, Copy)]
+#[derive(PartialEq, Debug, Clone, Copy)]
 pub enum Charset {
     UTF8,
     UTF8Mb4,
     Latin1,
     GBK,
+    Binary,
+    Ascii,
```

Comment:
Should  `Collation` at L107-116 be updated?

Reply:
nope, this two is the missing charset, just add them here is fine